mmdet/apis/ote/apis/detection/ote_utils.py

```diff
@@ -102,6 +103,21 @@ def on_train_batch_end(self, batch, logs=None):
         super().on_train_batch_end(batch, logs)
         self.update_progress_callback(self.get_progress())
 
+    def on_epoch_end(self, epoch, logs=None):
+        self.past_epoch_duration.append(time.time() - self.start_epoch_time)
+        self.__calculate_average_epoch()
+        score = None
+        if hasattr(self.update_progress_callback, 'metric') and isinstance(logs, dict):
+            score = logs.get(self.update_progress_callback.metric, None)
+        self.update_progress_callback(self.get_progress(), score)
+
+    def __calculate_average_epoch(self):
+        if len(self.past_epoch_duration) > self.epoch_history:
+            self.past_epoch_duration.remove(self.past_epoch_duration[0])
```

Comment:
IMO there are more explicit options, how to delete the first element from a list, e.g. `del self.past_epoch_duration[0]` or `self.past_epoch_duration.pop(0)`.

Reply:
Contributions to the SDK are always welcome.

Reply:
If you agree with the comment, could you please address it?

Reply:
Sure, I'm already trying to submit MR to impt.
https://gitlab.devtools.intel.com/vmc-eip/IMPT/impt/-/merge_requests/3921

Reply:
Ok, I see. You took the long road.

Reply:
I think I misunderstood your comment.
I updated it here.